test/core/event_engine/posix/dns_server.h

```diff
@@ -58,7 +61,7 @@ class DnsServer {
   grpc_core::Notification running_;
   mutable grpc_core::Mutex mu_;
   mutable grpc_core::CondVar cond_;
-  std::optional<std::array<uint8_t, 4>> ipv4_address_ ABSL_GUARDED_BY(mu_);
+  std::unordered_map<std::string, std::array<uint8_t, 4>> ipv4_addresses_ ABSL_GUARDED_BY(mu_);
```

Comment:
General recommendation to prefer absl::flat_hash_map when possible. See https://abseil.io/tips/136

You will then also be able to use string_view keys directly in test/core/event_engine/posix/dns_server.cc, instead of converting to a string.

Reply:
Those benchmarks exist, and are well documented. Search for SwissTable performance and design. The heterogeneous lookup is great too. That's why it's the standard recommended associative map type at Google when you don't need pointer stability. See node_hash_map otherwise.